cpp/src/arrow/scalar.cc

```diff
@@ -43,25 +43,31 @@ bool Scalar::Equals(const Scalar& other) const { return ScalarEquals(*this, othe
 StringScalar::StringScalar(std::string s)
     : StringScalar(Buffer::FromString(std::move(s))) {}
 
-FixedSizeBinaryScalar::FixedSizeBinaryScalar(const std::shared_ptr<Buffer>& value,
-                                             const std::shared_ptr<DataType>& type)
-    : BinaryScalar(value, type) {
-  ARROW_CHECK_EQ(checked_cast<const FixedSizeBinaryType&>(*type).byte_width(),
-                 value->size());
+FixedSizeBinaryScalar::FixedSizeBinaryScalar(std::shared_ptr<Buffer> value,
```

Comment:
In the future, can we limit the scope of PRs. Don't add stuff reported by clang-tidy of untouched code.